Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.h

```diff
@@ -40,9 +40,13 @@ namespace ScriptCanvas
     class ScriptCanvasRegistry
     {
     public:
+        virtual ~ScriptCanvasRegistry() = default;
         virtual void Init(NodeRegistry* nodeRegistry) = 0;
         virtual void Reflect(AZ::ReflectContext* context) = 0;
         virtual AZStd::vector<AZ::ComponentDescriptor*> GetComponentDescriptors() = 0;
+
+    protected:
+        AZStd::vector<AZ::ComponentDescriptor*> m_cachedDescriptors;
```

Comment:
Be careful, the AZ::SystemAllocator might not be available depend on where this AutoGen registry resides at.

Due to static destruction order, the AZ::Environment might have been detached at that point.

A way to avoid any of these issues, can be to cached the descriptors in an `AZStd::fixed_vector<AZ::ComponentDescriptors, 64>` instead.

Reply:
This is stuff that @onecent1101 added recently.  The lifetime of these ScriptCanvasRegistry objects is tied to that of modules related to Script Canvas and Script Canvas itself.  AZ::Environment and AZ::SystemAllocator should be available throughout.